src/gt4py/next/ffront/program_ast.py

```diff
@@ -72,12 +72,18 @@ class TupleExpr(Expr):
     elts: list[Expr]
 
 
+class AttributeExpr(Expr):
+    attr: str
+    value: Any
+    type: Optional[ts.TypeSpec] = None  # A003
+
+
 class Constant(Expr):
     value: Any  # TODO(tehrengruber): be more restrictive
 
 
 class Dict(Expr):
-    keys_: list[Name]
+    keys_: list[Union[Name, AttributeExpr]]
```

Comment:
is this needed? I would expect dict is nested instead of the key being an AttributeExpr.

Reply:
in `func_to_past` Dict keys can come in the form of Name or Attribute, respectively:
`domain={CellDim: (0, 10)}``
`domain={dims.CellDim: (0, 10)}`

Reply:
I see!